File: VIBEBackend/AvatarService/Models/AnalysisModel.cs
namespace AvatarService.Models;

public class AnalysisModel
{
	public long Id { get; set; }
	public string Scenario { get; set; }
	public string? Algorithm { get; set; }
	public string? Description { get; set; }
	public virtual ICollection<FileModel> Files { get; set; }

	public AnalysisDto Convert()
	{
		var dto = new AnalysisDto
		{
			Id = Id,
			Scenario = Scenario,
			Algorithm = Algorithm,
			Description = Description
		};

		List<FileDto>? files = new();
		if (Files != null)
			foreach (var item in Files)
			{
				files.Add(item.Convert());
			}

		dto.Files = files;
		return dto;
	}
}

File: frontend-vibe/src/app/models/analysis-model.ts
import FileModel from "./file-model";

export default class AnalysisModel {
  id: number | null;
  algorithm: string;
  scenario: string;
  files: FileModel[] = [];
  description: string;

  constructor(id: number | null = null, algorithm: string | null = null,
              scenario: string | null = null,
              description: string | null = null) {
    this.id = id;
    this.algorithm = algorithm;
    this.scenario = scenario;
    this.description = description;
  }
}
